src/api/spec/controllers/source_controller_spec.rb

```diff
@@ -50,4 +51,18 @@
     it { expect(response).to be_success }
     it { expect(project.config.to_s).to include('Updated', 'by', 'test') }
   end
+
+  describe "PUT #update_project_meta" do
+    it "raises exception for invalid DownloadRepository" do
+      login admin
+      bypass_rescue
+
+      xml = ActiveXML::Node.new(project.to_axml)
+      repo = xml.add_element 'repository', { name: "download_on_demand" }
+      repo.add_element 'download', { arch: "i586", url: "http://somewhere", repotype: "INVALID" }
+      request.env['RAW_POST_DATA'] = xml.dump_xml
+      expect { post :update_project_meta, params: { project: project } }.to raise_error(Project::SaveError, /Repotype is not included in the list/)
```

Comment:
Why not just let the controller run as it would in real life (without `bypass_rescue`) and assert on the response? i.e.
```ruby
request.env['RAW_POST_DATA'] = xml.dump_xml
post :update_project_meta, params: { project: project }

expect(response).to have_http_status(400)
expect(response.body).to match(/Repotype is not included in the list/)
```